subsys/bluetooth/controller/ll_sw/ll_addr.c

```diff
@@ -81,6 +60,31 @@ uint8_t ll_addr_set(uint8_t addr_type, uint8_t const *const bdaddr)
 	return 0;
 }
 
+uint8_t *ll_addr_get(uint8_t addr_type)
+{
+	if (addr_type > BT_ADDR_LE_RANDOM) {
+		return NULL;
+	}
+
+	if (addr_type) {
+		return rnd_addr;
+	}
+
+	return pub_addr;
+}
+
+uint8_t *ll_addr_read(uint8_t addr_type, uint8_t *const bdaddr)
+{
+	uint8_t *addr;
+
+	addr = ll_addr_get(addr_type);
+	if (addr) {
+		memcpy(bdaddr, addr, BDADDR_SIZE);
```

Comment:
Should we have an check or an ASSERT ensuring that `bdaddr` is non-NULL?

Reply:
As the parameter was passed by internal module caller (hci.c or ull_scan.c) a NULL can be avoided.